File: Swift-C++-IR-Comparison/test/functions/Swift/FunctionNestedFunction.swift
@main
struct MyMain {

    static func main() {
        var currentValue = 3
        let moveNearerToZero = chooseStepFunction(backward: currentValue > 0)
    }

}

func chooseStepFunction(backward: Bool) -> (Int) -> Int {
    func stepForward(input: Int) -> Int { return input + 1 }
    func stepBackward(input: Int) -> Int { return input - 1 }
    return backward ? stepBackward : stepForward
}

File: AutoTranspiledTests/try-0/FunctionNestedFunction.cpp
 

#include <iostream> 
 
typedef int (StepFunction) (int); 
  
int stepForward(int input) { 
    return input + 1; 
} 
  
int stepBackward(int input) { 
    return input - 1; 
} 
  
StepFunction* chooseStepFunction(bool backward) { 
    return backward ? stepBackward : stepForward; 
} 
  
int main() { 
    // main can be compiled in c++
    int currentValue = 3; 
    StepFunction* func = chooseStepFunction(currentValue > 0); 
  
    return 0; 
} 

  // End